extensions/Push/lib/Connector/Phabricator.pm

```diff
@@ -59,9 +54,32 @@ sub should_send {
 }
 
 sub send {
+    my ( $self, $message ) = @_;
+
     my $logger = Bugzilla->push_ext->logger;
 
-    $logger->info('AUDIT');
+    my $data = $message->payload_decoded;
+    my $bug_data = $self->_get_bug_data($data) || return 0;
+    my $bug = Bugzilla::Bug->new( { id => $bug_data->{id}, cache => 1 } );
+
+    my @synced_groups = split(/,/, Bugzilla->params->{phabricator_sync_groups});
+
+    if ( any { /$_->{name}/ ~~ @synced_groups } @{$bug->groups()} ) {
```

Comment:
What is '~~'?

Also we need to look at the groups that were added/removed as part of the bug change and see if any one of those are in @synced_groups. It will be this part in the event data:

   "event" : {
      "target" : "bug",
      "changes" : [
         {
            "removed" : "",
            "added" : "core-security, core-security-release",
            "field" : "bug_group"
         }
      ],
      ...
   }

You will need to split the groups on whitespace from 'added' and 'removed' and then combine them into a unique list. Then do compare the @changed_groups and @synced_groups to see if there are any intersected values. You can use the intersect() code from extensions/PhabBugz/lib/Util.pm.